src/components/modal/subject/create-subject.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from 'react'
+import { cva, VariantProps } from 'class-variance-authority'
+import { ModalHeader } from '../modal-components/modal-header'
+import { ModalRow } from '../modal-components/modal-row'
+import { ModalFooter } from '../modal-components/modal-footer'
+import { Input } from '../../input/input'
+
+import { SubjectCreateData } from '../../../data/models/subject.model'
+import Services from '../../../services'
+import '../modal.css'
+
+const createSubjectVariants = cva(
+  'base-modal input-modal',
+  {
+    variants: {
+      mode: {
+        light: 'light',
+        dark: 'dark'
+      }
+    },
+    defaultVariants: {
+      mode: 'light'
+    }
+  }
+)
+
+interface createSubjectProps extends VariantProps<typeof createSubjectVariants> {
+  className?: string
+  mode?: 'light' | 'dark'
+  variant?: 'solid' | 'outline'
+  confirm: Function
+  close: Function
+  mainSubject?: string
+  subjectID?: number
+  onAnimationEnd: Function
+}
+
+export function CreateSubject({ mode, variant, confirm, close, mainSubject, subjectID, className, onAnimationEnd }: createSubjectProps) {
+  const [ subjectData, setSubjectData ] = useState<SubjectCreateData>({
+    name: '',
+    mainSubject: mainSubject!
+  })
+
+  const handleChange = (e: any) => {
+    const { name, value } = e.target
+    setSubjectData((prevData) => ({ ...prevData, [name]: value }))
+  }
+
+  const handleSubmit = () => {
+    subjectID
+
+    ? Services.updateSubject(subjectID, subjectData)
+    .then(res => {
+      console.log(res)
+      confirm()
+      close()
+    })
+    .catch(err => { 
+      console.log(err) 
```

Comment:
So um ponto que talvez valha criar um TODO seria sobre tratamento de erros

Reply:
Real, a gente não tratou nenhum ainda no sistema web inteiro kk

Reply:
Mas eu ia puxar os tooltips do Pedrinho e mostrar pelo menos uma mensagem pro usuário no q não for tratável pelo menos